### PR TITLE
Add a SizeField attribute property that allows a field size to be directly read from another field

### PR DESCRIPTION
**Pull request by enclave-alistair**
Also did some internal renames, `IPacketProperty` to `IPacketField` to clarify when we're referring to a defined 'Field' within a packet, vs a 'Property' withing a type.

Updated README and added PacketFieldAttribute doc-comments.

### CODE REVIEW

**Comment by enclave-alistair** [comment]
@bcronje, would you be available to review this PR?

**Comment by bcronje** [comment]
For sure, I'll review this evening.

File: src/Enclave.FastPacket.Generator/FileTemplates/PacketFieldAttribute.cs [hunk 1]
```diff
@@ -19,17 +21,129 @@ public PacketFieldAttribute()
...
+        /// <summary>
+        /// Specify a fixed number of bytes that this field consumes in the packet.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Typically you would set this on a <see cref="ReadOnlySpan{T}"/> or <see cref="Span{T}"/> field,
+        /// to indicate the exact length of the buffer slice returned.
+        /// </para>
+        /// <para>
+        /// The start position of all fields following this one will be adjusted to accomodate this size.
```
Comment (bcronje) [inline src/Enclave.FastPacket.Generator/FileTemplates/PacketFieldAttribute.cs]: Typo - accomodate ? Unless that's the British spelling..

File: tests/Enclave.FastPacket.Generator.Tests/Helpers/CompilationVerifier.cs [hunk 2]
```diff
@@ -6,11 +6,43 @@
 
 namespace Enclave.FastPacket.Generator.Tests.Helpers;
 
-internal static class CompilationVerifier
+public class FluentVerify
```
Comment (bcronje) [inline tests/Enclave.FastPacket.Generator.Tests/Helpers/CompilationVerifier.cs:9]: Maybe change the file name as well?

**Comment by bcronje** [comment]
Apart from the two very minor comments, looks good to me.  This will be a very useful feature, thank you @enclave-alistair

**Comment by enclave-alistair** [comment]
Feedback applied @bcronje, thanks.

**Pull request merged**